main.cpp

```diff
@@ -36,60 +36,47 @@ T::T(int v, const char* p): value(v), name(p) {}
 
 struct Comparator                               //4
 {
-    T* compare(T* a, T* b) //5
+    T* compare(T& a, T& b) //5
     {
-        if (a != nullptr && b != nullptr)
-        {
-            if( a->value < b->value ) return a;
-            if( a->value > b->value ) return b;
-        }        
+        if( a.value < b.value ) return &a;
+        if( a.value > b.value ) return &b;     
         return nullptr;
     }
 };
 
 struct U
 {
     float x { 0 }, y{ 0 };
-    float computeDistance(float* xUpdated)      //12
+    float computeDistance(const float &xUpdated)      //12
```

Comment:
put the `&` next to the type.   when you put it next the variable name, it looks like you're trying to get the address of the variable.
- [ ] revise